spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkSessionCatalog.java

```diff
@@ -50,8 +53,8 @@
  *
  * @param <T> CatalogPlugin class to avoid casting to TableCatalog and SupportsNamespaces.
  */
-public class SparkSessionCatalog<T extends TableCatalog & SupportsNamespaces> extends BaseCatalog
-    implements CatalogExtension {
+public class SparkSessionCatalog<T extends TableCatalog & FunctionCatalog & SupportsNamespaces>
```

Comment:
Guard is required in `setDelegateCatalog`

Reply:
And the doc should also be updated.

Reply:
Thanks, both suggestions are addressed.

Reply:
Why is this changed? BaseCatalog Supports FunctionCatalog already
https://github.com/apache/iceberg/blob/aaa67d0797a59301a57c8603b8a22852e67e4f05/spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/BaseCatalog.java#L41

Reply:
Here makes the class T extends FunctionCatalog, and it is for the delegated session catalog (set via Sparks's `CatalogPlugin#setDelegateCatalog`), but not for the Iceberg's SparkSessionCatalog itself. This PR is mainly purposed for loading functions from the delegated session catalog, therefore, forcing to check it to be an instance of (or as) FunctionCatalog is necessary.